IviDriverCore/1.0/Spec/IviDriverCore.md

```diff
@@ -0,0 +1,770 @@
+
+For example:
+
+- Initialization may be implemented as a constructor in .NET, but might be implemented as a method in another language.
+
+- Supported Instrument Models is implemented as a method called `GetSupportedInstrumentModels()` in .NET, but might be implemented as a property in another language.
+
+Refer to language-specific specifications for implementation details.
+
+### Initialization (Construction)
+
+Depending on the language, IVI Core Drivers may be initialized in a constructor or in a method. This process is called *initialization* in this section.
+
+The following parameters are defined for the *initialization* operation:
+
+- ***Resource Name*:**
+  An instrument specific string that identifies the address of the instrument, such as a VISA resource descriptor string.
+
+- ***ID Query*:**
+  Specifies if the identity of the instrument should be verified during *initialization*.
+
+- ***Reset*:**
+  Specifies if the instrument should be reset during *initialization*.
+
+- ***Simulation*:**
+  Drivers shall support a way to specify if the driver is placed in simulation mode during initialization. By default Simulate shall be false.
+
+> **Observation:**
+> > Drivers may implement multiple initializations that include different subsets of the parameters described here, or other parameters defined by the driver.
+
+### Driver Version
+
+Provides a driver version string. This is a version optionally followed by a descriptive string.
+
+The format of the version shall follow the rules for FileVersion defined in [File Versioning](#file-versioning) followed by an optional string. If the string is present, a space shall separate the version from the string. The string contains additional driver specific version information. Multi-byte characters are not allowed in the string that this property returns. String characters shall be in the range of `\x20` - `\x7E`.
+
+Examples of allowed version strings are shown below:
+
+    4.00.1
+
+    02.0001.12345.1 This version adds XYZ capability to the component
+
+### Driver Vendor
+
+Returns the name of the vendor that supplies the IVI Core driver.
+
+### Error Query
+
+The Error Query member queries the instrument and returns instrument specific error information.
+
+For instruments that implement an error queue, Error Query extracts a single error from the queue.
+
+For instruments that have status registers but no error queue, the IVI driver returns an error consistent with instrument design.
+
+The operation of Error Query is not affected by the value of Query Instrument Status Enabled.
+
+### Instrument Manufacturer
+
+Returns the name of the manufacturer of the instrument. The IVI driver returns the value it queries from the instrument or a string indicating that it cannot query the instrument identity. For instance, "Cannot query from instrument".
```

Comment:
In the case in which a value cannot be queried from the instrument, wouldn't it make more sense and be less error-prone to clients to return an error code?